### PR TITLE
Duplicate username signup patch

### PR DESCRIPTION
**Pull request by Kapn**
When signing up, if a username exists, append the next available # to the end of the username.

i.e:

1. signing up with kp@gmail.com will create user w/ username 'kp'
2. signing up with kp@yahoo.com will create user w/ username 'kp1'
3. signing up with kp@aol.com will create user w/ username 'kp2'
.
.
.

### CODE REVIEW

File: teamwork/apps/profiles/views/BaseView.py [hunk 1]
```diff
@@ -9,16 +9,47 @@
 # View Imports
 from teamwork.apps.profiles.views.EditProfileView import edit_profile
 
+def find_available_username(username):
+    """
+    checks if the passed username is available to be used.
+    If not, then return the 'username + #' available
```
Comment (CoolCatMona) [inline teamwork/apps/profiles/views/BaseView.py]: Now may be a good time to implement better documentation, especially with some simple stuff

i.e
:param str username: User's name.

File: teamwork/apps/profiles/views/BaseView.py [hunk 2]
```diff
@@ -9,16 +9,47 @@
...
+        return "{}{}".format(username, value + 1)
+
+    return username
+
+def parse_username_num(user, username_to_parse):
+    """
+    Parses out the number at the end of a user's username given the base username w/o the number
+    """
+    try:
+        return int(user.username[len(username_to_parse):])
```
Comment (CoolCatMona) [inline teamwork/apps/profiles/views/BaseView.py]: Neat way to get # from username!

**Review by CoolCatMona — APPROVED** [review]
Thanks for including tests!
If possible, could you add some parameters to the docstrings you created?
No sweat if not, good to go

File: teamwork/apps/profiles/views/BaseView.py (hunk 1 above)
Comment (Kapn) [reply teamwork/apps/profiles/views/BaseView.py]: @SeanmanX Will do. I'm not familiar w/ the most popular styling for this in python. I used this for reference:
[Google Style Python Docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)

**Pull request merged**